js/common/lib/training-session.ts

```diff
@@ -49,21 +50,29 @@ export interface TrainingSession {
   // #endregion
 
   // #region copy parameters
+
+  /**
+   * Retrieves the size of all parameters for the training state.
+   *
+   * @param trainableOnly skips non-trainable parameters when true.
+   */
+  getParametersSize(trainableOnly: boolean): Promise<number>;
+
   /**
    * Copies from a buffer containing parameters to the TrainingSession parameters.
    *
    * @param buffer - buffer containing parameters
    * @param trainableOnly - True if trainable parameters only to be modified, false otherwise.
    */
-  loadParametersBuffer(array: Uint8Array, trainableOnly: boolean): Promise<void>;
+  loadParametersBuffer(array: Float32Array, trainableOnly: boolean): Promise<void>;
 
   /**
    * Copies from the TrainingSession parameters to a buffer.
    *
    * @param trainableOnly - True if trainable parameters only to be copied, false othrwise.
    * @returns A promise that resolves to a buffer of the requested parameters.
```

Comment:
When set to true only trainable parameters (parameters for which requires_grad is set to true) are copied.